### ISSUE TITLE
Improve goldchain JumpScaleX code

### ISSUE THREAD

**Issue by GlenDC** [nbh-digital/goldchain#30]
- [ ] ensure that JSX tfchain client provides multichain support, such that network types are linked to a chain type, and that the chain type also defines configuration around unit/precision (JSX issue: https://github.com/threefoldtech/jumpscaleX/issues/730)
- [x] support auth coin transactions (JSX issue: https://github.com/threefoldtech/jumpscaleX/issues/731)
- [x] support mint destruction transaction (JSX issue: https://github.com/threefoldtech/jumpscaleX/issues/732)

**Comment by LeeSmet** [comment]
https://github.com/threefoldtech/jumpscaleX/issues/799

**Comment by LeeSmet** [comment]
I think the best way forward for now is to copy the relevant pieces of tfchain code to a new goldchain wallet. This will allow us to implement the methods and unblock the gateway issue. Already started implementing the coin destruction tx for this, but not yet finished. I'll copy the code and push it to a branch on jsx later. After that, the issue is not blocking and we can take our time to properly refactor the 2 clients to a stand alone base rivine client

**Comment by LeeSmet** [comment]
Update: goldchain client has been added, and coin destruction tx inital code is added as well. Now working on trying to fix some of the tests. It seems that coin destruction tx also still has some bugs, however the rivine documentation about the binary encoding gives a wrong example itself, and there are no tests for this transaction it seems (that  I found), so will need to reverse engineer this a bit to get it working properly

**Comment by LeeSmet** [comment]
added support for auth address update transactions.  This restores wallet functionality, which was broken as all wallets which have outputs contain at least one auth address update tx, which previously could not be decoded (unless you are using a genesis output, but even then it could not spend)

**Comment by LeeSmet** [comment]
auth condition update txes have also been added.

Still need to update the explorer client to support the auth condition endpoints, and create some kind of auth tx "mini client" to add to the wallets for user friendly interactions with these transactions

**Comment by GlenDC** [comment]
Resolved, remainder of planned work in this issue was no longer relevant.
